duo-client/src/main/java/com/duosecurity/client/Http.java

```diff
@@ -257,16 +261,18 @@ public void useCustomCertificates(String[] customCaCerts) {
   }
 
   protected String canonRequest(String date, int sigVersion)
-      throws UnsupportedEncodingException {
-    String canon = "";
-    if (sigVersion == 2) {
-      canon += date + "\n";
+      throws UnsupportedEncodingException, IllegalArgumentException {
+    String canon;
+    String queryString = createQueryString();
+    if (sigVersion == 1){
+      canonicalizer = new v1();
+      canon = canonicalizer.canonicalize(method, host, uri, queryString);
+    } else if (sigVersion == 2){
+      canonicalizer = new v2(date);
+      canon = canonicalizer.canonicalize(method, host, uri, queryString);
+    } else {
+      throw new IllegalArgumentException("Invalid Request Signature");
```

Comment:
Not sure if this should be a custom exception or `IllegalArgumentException` is good enough for us. I think the name of this exception make sense to me in this check.